kitura-init.js

```diff
@@ -1,32 +1,150 @@
 #!/usr/bin/env node
 
 const chalk = require('chalk');
-const spawn = require('child_process').spawn;
+const spawnSync = require('child_process').spawnSync;
+var os = require('os');
+var fs = require('fs');
 
 let args = process.argv.slice(2);
 
-let options = ['-p', 'yo@1', '-p', 'generator-swiftserver', '--', 'yo', 'swiftserver'];
-
 if (args.length > 0) {
     if (args.indexOf('--help') > -1) {
         printHelp();
         process.exit(0);
     }
+}
+
+let path = require("path");
+let currentDirPath = path.resolve("./");
+let currentDir = path.basename(currentDirPath);
+// Replace spaces with hyphens so Xcode project will build.
+let projName = currentDir.replace(/ /g,"-")
+
+if (projName.charAt(0) === '.') {
+     console.error('Application name cannot start with .: %s')
+   }
+   if (projName.toLowerCase() === 'node_modules') {
+    console.error('Application name cannot be {node_modules}')
+   }
+   if (projName.toLowerCase() === 'favicon.ico') {
+     console.error('Application name cannot be {favicon.ico}')
+}
+
+// Make sure directory is empty.
+checkEmptyDir()
+
+// Clone repo contents into current directory.
+cloneProject()
+
+// Rename project to match current directory
+renameProject()
+
+// If '--skip-build' not specified, then build project.
+if (!(args.includes('--skip-build'))) {
+    buildProject()
+}
+
+function checkEmptyDir() {
+  try {
+    var data = fs.readdirSync('.')
+    if (data.length !== 0) {
+      console.error(chalk.red('Current directory is not empty.'));
+      console.error(chalk.red('Please repeat the command in an empty directory.'));
+      process.exit()
+    }
 
-    // Add on the arguments
-    options.push(args);
+  } catch(err) {
+    console.error(err)
+    process.exit()
+  }
 }
 
-options.push('--init');
+function cloneProject() {
+  console.log('Creating project...')
+  let clone = spawnSync('git', ['clone', '-b', 'init', 'https://github.com/IBM-Swift/generator-swiftserver-projects', '.'])
+
+  if (clone.status !== 0) {
+    console.error(chalk.red('Error: ') + 'failed to run git clone.');
+    console.error('Please check that you have git installed.');
+    console.error('Head to https://git-scm.com/downloads for more info.');
+    process.exit(clone.status);
+  }
+  else {
+    console.log(chalk.green('Project created successfully.'));
+    if ((args.includes('--skip-build'))) {
+      console.log('Next steps:')
+      console.log('')
+      console.log('Generate your Xcode project:')
+      console.log(chalk.grey('   $ swift package generate-xcodeproj'))
+      console.log('')
+      console.log('Or, build your project from the terminal:')
+      console.log(chalk.grey('   $ swift build -Xlinker -lc++'))
+    }
+  }
+
+  // Remove git tracking
+  let git = spawnSync('rm', ['-rf', '.git'])
+  if (git.status !== 0) {
+    console.error(chalk.red('Error: ') + 'failed to remove git tracking.');
```

Comment:
I think "tracking" implies something to do with us tracking usage, which we don't do. Perhaps `failed to remove .git directory` is better.